src/pages/cartDetail/index.js

```diff
@@ -0,0 +1,84 @@
+import { useRef } from "react";
+import { useHistory } from "react-router";
+import { useEffect, useState } from "react/cjs/react.development";
+import CartItem from "../../components/cartItem";
+import { ErrorField } from "../../components/elements";
+import { HOMEPAGE } from "../../constants";
+import {
+  getCartItems,
+  deleteCartItem,
+  updateCart,
+  checkoutCart,
+} from "../../services/cartService";
+import "./cartDetail.css";
+
+export default function CartDetail() {
+  const [cartItems, setCartItems] = useState([]);
+  const [errorMessage, setErrorMessage] = useState("Cart is Empty");
+  const formRef = useRef();
+  const history = useHistory();
+  const updateCartItems = () => {
+    getCartItems()
+      .then((cartItems) => {
+        setCartItems(cartItems);
+      })
+      .catch((errors) => {
+        setCartItems([]);
+        setErrorMessage(errors);
+      });
+  };
+  const onChangeHandler = (event) => {
+    const formData = new FormData(formRef.current);
+    if (event.target.value !== "") {
+      updateCart(formData);
+    }
+  };
```

Comment:
Updating the cart on each input change doesn't seem like a good user experience. If a user wants to change quantity from 1 to 2, this will be invoked twice once when backspace will be pressed to remove 1 and the second time when 2 will be typed. This can be optimized.